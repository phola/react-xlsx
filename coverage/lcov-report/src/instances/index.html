<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/instances</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/instances
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>20/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>20/25</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="86.96" class="pic high"><div class="chart"><div class="cover-fill" style="width: 86%;"></div><div class="cover-empty" style="width:14%;"></div></div></td>
	<td data-value="86.96" class="pct high">86.96%</td>
	<td data-value="23" class="abs high">20/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="86.96" class="pct high">86.96%</td>
	<td data-value="23" class="abs high">20/23</td>
	</tr>

<tr>
	<td class="file low" data-value="subtest.js"><a href="subtest.js.html">subtest.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Sep 16 2016 08:43:05 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
