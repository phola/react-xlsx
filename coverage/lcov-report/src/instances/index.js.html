<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/instances/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/instances</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/7</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import WorkBook from '../WorkBook'
import XCell from '../XCell'
import Sheet from '../Sheet'
import XTrigger from '../XTrigger'
import FileSaver from 'file-saver'
&nbsp;
const defaultCellStyle = { font: { name: 'Verdana', sz: 11, color: 'FF00FF88' }, fill: { fgColor: { rgb: 'FFFFAA00' } } }
&nbsp;
const handleXLSX = <span class="fstat-no" title="function not covered" >(b</span>lob) =&gt; {
<span class="cstat-no" title="statement not covered" >  console.log(blob)</span>
<span class="cstat-no" title="statement not covered" >  debugger</span>
<span class="cstat-no" title="statement not covered" >  FileSaver.saveAs(blob, 'test.xlsx')</span>
}
&nbsp;
export const simple = () =&gt; {
  return (&lt;WorkBook defaultCellStyle={defaultCellStyle} onXLSXGenerated={handleXLSX}&gt;
            &lt;Sheet name='woooo'&gt;
              &lt;XCell
                row={0}
                col={1}
                colSpan={2}
                rowSpan={6}&gt;
                &lt;div&gt;
                  &lt;div&gt;
                    nested
                  &lt;/div&gt;
                &lt;/div&gt;
              &lt;/XCell&gt;
            &lt;/Sheet&gt;
            &lt;Sheet name='hoooo'&gt;
              &lt;XCell row={1} col={2} cellStyle={{font: {bold: true}}}&gt;
                I am bold
              &lt;/XCell&gt;
              &lt;XCell cellRef='C9'&gt;
                plain string
              &lt;/XCell&gt;
            &lt;/Sheet&gt;
            &lt;XTrigger action='onClick'&gt;
              &lt;button&gt;
                get XLSX
              &lt;/button&gt;
            &lt;/XTrigger&gt;
          &lt;/WorkBook&gt;)
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Sep 08 2016 09:35:04 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
