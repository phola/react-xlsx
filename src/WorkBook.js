import React, { Component } from 'react'
import XLSX from 'xlsx-style'
import s2ab from './s2ab'
import { groupBy } from 'lodash'

// note this component goes against typical react practice of one way top down data flow to allow
// the user to arbitrarly place sheets and cell components anywhere in the component heirarchy.
// the ancestor components broadcast values up to the WorkBook via context.

export class WorkBook extends Component {

  constructor (props) {
    super(props)
    this.state = { cells: [] }
  }

  getChildContext () {
    return {
      getCell: this.getCell.bind(this),
      renderCells: this.props.renderCells,
      toXLSX: this.toXLSX.bind(this)
    }
  }

  getCell (val) {
    var { cells } = this.state
    var cell = cells.find(c => c.cellRef === val.cellRef)
    if (cell) {
      cell = val
    }else {
      cells.push(val)
    }
    this.setState({ cells: cells })
  }

  updateRange (row, col, range) {
    if (range.s.r > row) { range.s.r = row }
    if (range.s.c > col) { range.s.c = col }
    if (range.e.r < row) { range.e.r = row }
    if (range.e.c < col) { range.e.c = col }
    return range
  }

  mapCells (cells) {
    var mapped = {}
    var merges = []
    if (cells.length > 0) {
      var range = {s: {c: 10000000, r: 10000000}, e: {c: 0, r: 0}}
      cells.forEach(cell => {
        var { cellRef, col, row, data, merge } = cell
        range = this.updateRange(row, col, range)
        if (merge) {
          merges.push(merge)
        }
        mapped[cellRef] = data
      })
      mapped = Object.assign(mapped, {'!ref': XLSX.utils.encode_range(range)}, {'!merges': merges})
    }
    return mapped
  }

  toJSON (cb) {
    const { children, title, author } = this.props
    var wb
    var sheets = groupBy(this.state.cells, 'sheet')
    if (sheets) {
      wb = { Sheets: {}, SheetNames: [] }
      Object.keys(sheets).forEach(sheet => {
        wb.SheetNames.push(sheet)
        wb.Sheets[sheet] = this.mapCells(sheets[sheet])
      }
      )
    }
    if (cb) {
      cb(wb)
    }
    else {
      return wb
    }
  }

  writeXLSX (wb) {
    const { defaultCellStyle } = this.props
    var wopts = { bookType: 'xlsx', bookSST: false, type: 'binary', defaultCellStyle: defaultCellStyle }
    return XLSX.write(wb, wopts)
  }

  toXLSX (wb = this.toJSON()) {
    var blob = new Blob([s2ab(this.writeXLSX(wb))], {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    if (this.props.toXLSXCallback) {
      this.props.toXLSXCallback(blob)
    }
    else {
      return blob
    }
  }

  render () {
    const { children, generateXLSX = false, toXLSXCallback } = this.props

    if (generateXLSX && toXLSXCallback) this.toXSLX()

    return <div>{children}</div>

  }
}

WorkBook.propTypes = {
  title: React.PropTypes.string,
  author: React.PropTypes.string,
  renderCells: React.PropTypes.bool,
  toXLSXCallback: React.PropTypes.func,
  toJSONCallback: React.PropTypes.func,
  children: React.PropTypes.oneOfType([
    React.PropTypes.object,
    React.PropTypes.array
  ])
}

WorkBook.defaultProps = {
  title: 'generated by react-xlsx',
  author: 'react-xlsx',
  renderCells: true
}

WorkBook.childContextTypes = {
  getCell: React.PropTypes.func,
  toXLSX: React.PropTypes.func,
  renderCells: React.PropTypes.bool
}
