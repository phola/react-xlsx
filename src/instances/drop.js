import { workBookFromFile } from '../index'
import React, { PropTypes, Component } from 'react'
import Dropzone from 'react-dropzone'
export default class Drop extends React.Component {
  constructor () {
    super()
    this.state = { files: [] }
  }

  onDrop (files) {
    const updateState = file => {
      this.setState({
        files: [...this.state.files, ...[file]]
      })
    }
    files.forEach(file => {
      workBookFromFile(file, w => {
        file.Workbook = w
        updateState(file)
      })
    })
  }

  render () {
    return (
      <section>
        <div className='dropzone'>
          <Dropzone onDrop={this.onDrop.bind(this)}>
            <p>
              Try dropping some files here, or click to select files to upload.
            </p>
          </Dropzone>
        </div>
        <aside>
          <h2>Dropped files</h2>
          <ul>
            {this.state.files.map(f => (
              <li>
                {f.name} - {f.size} bytes
                <ul>
                  <h3>Sheets</h3>
                  {f.Workbook.SheetNames.map(s => (
                    <li>
                      {s}
                    </li>
                  ))}
                </ul>
              </li>
            ))}
          </ul>
        </aside>
      </section>
    )
  }
}
