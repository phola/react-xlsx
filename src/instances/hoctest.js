import React, { PropTypes, Component } from 'react'
import { WorkBookHOC, WorkBook, Sheet, Cell, XTrigger } from '../index'
import SubTest from './subtest'
import FileSaver from 'file-saver'

class HocTest extends Component {
  render () {
   return (<div>
            <Sheet name='hoooo'>
            <SubTest cellRef='C9' content='feesh'/>
            <SubTest cellRef='C10' content='feesh'/>
            </Sheet>
            <Sheet name='boooo'>
            <SubTest cellRef='C9' content='foossh'/>
            <SubTest cellRef='C10' content='foosh'/>
            </Sheet>
            </div>)
  }
}

const handleXLSX = (blob) => {
          console.log(blob)
          FileSaver.saveAs(blob, 'formulae.xlsx')
        }

export default WorkBookHOC(HocTest, {render: false, debug: false, generate: true, callback: handleXLSX})
