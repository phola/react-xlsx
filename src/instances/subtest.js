import React, { PropTypes, Component } from 'react'
import { Cell } from '../index'

export default class SubTest extends Component {
  render () {
  const {content, cellRef} = this.props
  return <Cell cellRef={cellRef}>
            {content}
          </Cell>
  }
}
