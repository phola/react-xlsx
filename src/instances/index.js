import React from 'react'
import WorkBook from '../WorkBook'
import XCell from '../XCell'
import Sheet from '../Sheet'
import XTrigger from '../XTrigger'
import FileSaver from 'file-saver'

const defaultCellStyle = { font: { name: 'Verdana', sz: 11, color: 'FF00FF88' }, fill: { fgColor: { rgb: 'FFFFAA00' } } }

const handleXLSX = (blob) => {
  console.log(blob)
  debugger
  FileSaver.saveAs(blob, 'test.xlsx')
}

export const simple = () => {
  return (<WorkBook defaultCellStyle={defaultCellStyle} onXLSXGenerated={handleXLSX}>
            <Sheet name='woooo'>
              <XCell
                row={0}
                col={1}
                colSpan={2}
                rowSpan={6}>
                <div>
                  <div>
                    nested
                  </div>
                </div>
              </XCell>
            </Sheet>
            <Sheet name='hoooo'>
              <XCell row={1} col={2} cellStyle={{font: {bold: true}}}>
                I am bold
              </XCell>
              <XCell cellRef='C9'>
                plain string
              </XCell>
            </Sheet>
            <XTrigger action='onClick'>
              <button>
                get XLSX
              </button>
            </XTrigger>
          </WorkBook>)
}
