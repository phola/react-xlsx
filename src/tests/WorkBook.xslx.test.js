import React from 'react'
import WorkBook from '../WorkBook'
import { simple } from '../instances'
import ReactDOM from 'react-dom'
import TestUtils from 'react-addons-test-utils'

it('generates simple workbook object', () => {
  const jsx = simple()
  const component = TestUtils.renderIntoDocument(
    jsx
  )
  let tree = component.createWorkBook()
  expect(tree).toMatchSnapshot()

  tree = { blobsize: component.generateXLSX(tree).size }
  expect(tree).toMatchSnapshot()
})
