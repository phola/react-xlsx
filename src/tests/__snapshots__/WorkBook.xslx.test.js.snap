exports[`test generates simple workbook object 1`] = `
Object {
  "Props": Object {
    "Author": "react-xlsx",
    "Title": "generated by react-xlsx"
  },
  "SheetNames": Array [
    "woooo",
    "hoooo"
  ],
  "Sheets": Object {
    "hoooo": Object {
      "!merges": Array [],
      "!ref": "C2:C9",
      "C2": Object {
        "s": Object {
          "font": Object {
            "bold": true
          }
        },
        "t": "s",
        "v": "I am bold"
      },
      "C9": Object {
        "s": Object {},
        "t": "s",
        "v": "plain string"
      }
    },
    "woooo": Object {
      "!merges": Array [
        Object {
          "e": Object {
            "c": 2,
            "r": 5
          },
          "s": Object {
            "c": 1,
            "r": 0
          }
        }
      ],
      "!ref": "B1",
      "B1": Object {
        "s": Object {},
        "t": "s",
        "v": "nested"
      }
    }
  }
}
`;

exports[`test generates simple workbook object 2`] = `
Object {
  "blobsize": 15266
}
`;
